api/v1alpha1/applicationset_types.go

```diff
@@ -158,6 +158,7 @@ type GitFileGeneratorItem struct {
 type SCMProviderGenerator struct {
 	// Which provider to use and config for it.
 	Github *SCMProviderGeneratorGithub `json:"github,omitempty"`
+	Gitlab *SCMProviderGeneratorGitlab `json:"gitlab,omitempty"`
 	// TODO other providers.
```

Comment:
We can probably remove this TODO now that we have handled the two largest providers, but PRs are still welcome for others if anyone else in the community is interested :smiley:

Reply:
removed